### ISSUE TITLE
Using YOLO26n for object detection, under data imbalance, the model's performance dropped by half after changing BCEWithLogitsLoss to FocalLoss

### ISSUE THREAD

**Issue by wang-yu-tao** [ultralytics/ultralytics#24004]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train, Val

### Bug

Using YOLO26n for object detection, under data imbalance, the model's performance dropped by half after changing BCEWithLogitsLoss to FocalLoss.

Under v8DetectionLoss class in ultralytics\utils\loss.py 
self.bce = nn.BCEWithLogitsLoss ( reduction = ' none ' ) 
Change to 
self.bce = FocalLoss ( gamma = 1.5, alpha = 0.25 )

### Environment

Windows10,
ultralytics 8.4.6,
python3.11.5,
RAM 32GB,
24GB RTX3090，
torch 2.9.1+cu126

### Minimal Reproducible Example

import os
from ultralytics import YOLO
from ultralytics.utils import SETTINGS
import warnings

warnings.filterwarnings("ignore", category=FutureWarning)
warnings.filterwarnings("ignore", category=UserWarning)

if __name__ == '__main__':
    os.environ['WANDB_DISABLED'] = 'true'  #  wandb
    SETTINGS['tensorboard'] = True  #  TensorBoard

    model = YOLO("yolo26.yaml").load('yolo26n.pt')

    model.train(data="my-rsd.yaml", epochs=200, batch=32, lr0=0.001, optimizer='AdamW', patience=50, imgsz=640, workers=8, device=0,  # resume=True,  )

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @wang-yu-tao, thank you for reporting this issue with YOLO26 🚀 This is an automated response to help gather the right debugging details, and an Ultralytics engineer will also assist you soon.

We recommend a quick visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, along with answers to many common questions.

Since this is a 🐛 Bug Report, thank you for already sharing a reproducible snippet. To help us investigate the loss-function change more efficiently, please also provide a complete [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) if possible, including:
- The exact `ultralytics/utils/loss.py` modification you applied
- Training logs for both the default loss and the modified `FocalLoss` run
- Validation metrics showing the reported performance drop
- A small dataset sample or a simplified setup that reproduces the behavior

If this is also related to custom training behavior ❓, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧 Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community 🤝

## Upgrade

Please upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify this issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by glenn-jocher** [comment]
This is expected rather than a bug: `v8DetectionLoss` is tuned around `nn.BCEWithLogitsLoss(reduction="none")`, so swapping in `FocalLoss` changes the classification optimization enough to hurt YOLO26 detection performance; for imbalanced datasets, the safer approach is to keep the default loss and add per-class weights with a custom trainer as shown in the [custom loss guide](https://docs.ultralytics.com/guides/custom-trainer/) and the [`v8DetectionLoss` reference](https://docs.ultralytics.com/reference/utils/loss/).
